core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -4497,7 +4514,10 @@ protected function parentView()
 
 			for ($i=0, $c=\count($row); $i<$c; $i++)
 			{
-				$this->denyAccessUnlessGranted(ContaoCorePermissions::DC_ACTION_VIEW, new DataContainerSubject($this->strTable, $row[$i]['id']));
+				// Improve performance
+				$this->setCurrentRecordCache($row[$i]['id'], $this->strTable, $row[$i]);
```

Comment:
`$row[$i]` might contain an additional `foreignKey` column here because of line 4439, so it should not be used as the current record.

However, this can be fixed in Line 4439 by moving the subquery to the `ORDER BY` clause and keeping `$query` untouchted:

```php
$key = explode('.', $GLOBALS['TL_DCA'][$this->strTable]['fields'][$firstOrderBy]['foreignKey'], 2);
$orderBy[0] = "(SELECT " . Database::quoteIdentifier($key[1]) . " FROM " . $key[0] . " WHERE " . $this->strTable . "." . Database::quoteIdentifier($firstOrderBy) . "=" . $key[0] . ".id)";
```

Reply:
I noticed that too but ignored it :P Fixed according to your suggestion.